File: src_original/extras-kwidgetsaddons/libkpassword.cpp
#include <KPassword>
#include <QMetaObject>
#include <kpassword.h>
#include "libkpassword.h"
#include "libkpassword.hxx"

QMetaObject* KPassword_GetEnumMetaObject(int param1) {
    return (QMetaObject*)KPassword::qt_getEnumMetaObject(static_cast<KPassword::RevealMode>(param1));
}

const char* KPassword_GetEnumName(int param1) {
    return (const char*)KPassword::qt_getEnumName(static_cast<KPassword::RevealMode>(param1));
}

File: include_original/extras-kwidgetsaddons/libkpassword.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const kpassword_enums = enums;
const std = @import("std");

/// https://api.kde.org/kpassword.html
pub const kpassword = struct {
    /// [Qt documentation](https://api.kde.org/kpassword.html#qt_getEnumMetaObject)
    ///
    /// ``` param1: kpassword_enums.RevealMode ```
    pub fn GetEnumMetaObject(param1: i32) QtC.QMetaObject {
        return qtc.KPassword_GetEnumMetaObject(@intCast(param1));
    }

    /// [Qt documentation](https://api.kde.org/kpassword.html#qt_getEnumName)
    ///
    /// ``` param1: kpassword_enums.RevealMode ```
    pub fn GetEnumName(param1: i32) []const u8 {
        const _ret = qtc.KPassword_GetEnumName(@intCast(param1));
        return std.mem.span(_ret);
    }
};

/// https://api.kde.org/kpassword.html#types
pub const enums = struct {
    pub const RevealMode = enum {
        pub const OnlyNew: i32 = 0;
        pub const Never: i32 = 1;
        pub const Always: i32 = 2;
    };
};
